lxd/db/node.go

```diff
@@ -104,6 +106,11 @@ func (n NodeInfo) ToAPI(cluster *Cluster, node *Node, leader string) (*api.Clust
 			return errors.Wrap(err, "Get max version")
 		}
 
+		groups, err = tx.GetClusterGroupsWithNode(n.Name)
```

Comment:
I'm confused why we need to populate groups again in this function, when it should have already been populated into the `NodeInfo.Groups` field by the the `nodes()` function, can we not just copy it from the `n.Groups` field?